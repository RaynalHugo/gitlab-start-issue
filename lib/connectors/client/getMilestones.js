"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));

var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));

var _fp = require("lodash/fp");

function getMilestones() {
  return _getMilestones.apply(this, arguments);
}

function _getMilestones() {
  _getMilestones = (0, _asyncToGenerator2.default)(
  /*#__PURE__*/
  _regenerator.default.mark(function _callee() {
    var client, store, projectId, milestones;
    return _regenerator.default.wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            client = this.client;
            store = this.store;

            if (!(0, _fp.isNil)(client)) {
              _context.next = 6;
              break;
            }

            throw new Error("no client in 'getMilestones'");

          case 6:
            if (!(0, _fp.isNil)(store)) {
              _context.next = 10;
              break;
            }

            throw new Error("no store in 'getMilestones'");

          case 10:
            _context.next = 12;
            return store.getItem("projectId");

          case 12:
            projectId = _context.sent;
            _context.t0 = _fp.get;
            _context.next = 16;
            return client.get("/projects/".concat(projectId, "/milestones"));

          case 16:
            _context.t1 = _context.sent;
            milestones = (0, _context.t0)("data", _context.t1);
            return _context.abrupt("return", milestones);

          case 19:
          case "end":
            return _context.stop();
        }
      }
    }, _callee, this);
  }));
  return _getMilestones.apply(this, arguments);
}

var _default = getMilestones;
exports.default = _default;