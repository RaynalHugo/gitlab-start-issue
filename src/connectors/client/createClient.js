import axios from "axios";
import { isNil } from "lodash/fp";

export function createClient({ userPrivateToken, gitlabUrl }) {
	if (!isNil(userPrivateToken)) {
		return axios.create({
			baseURL: gitlabUrl,
			timeout: 10000,
			headers: { "Private-Token": userPrivateToken }
		});
	} else {
		throw new Error(
			"userPrivateToken is not definied in client `createClient`"
		);
	}
}

export const client = createClient({
	gitlabUrl: "https://gitlab.com/api/v4",
	userPrivateToken: "ddx1wWi8xo6ryc4XbBqj"
});
