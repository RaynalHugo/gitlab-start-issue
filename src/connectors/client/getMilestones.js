import { get, isNil } from "lodash/fp";

async function getMilestones() {
	const client = this.client;
	const store = this.store;
	if (isNil(client)) {
		throw new Error("no client in 'getMilestones'");
	} else if (isNil(store)) {
		throw new Error("no store in 'getMilestones'");
	} else {
		const projectId = await store.getItem("projectId");
		const milestones = get(
			"data",
			await client.get(`/projects/${projectId}/milestones`)
		);
		return milestones;
	}
}

export default getMilestones;

async function getMilestones2(client, { projectId }) {
	const store = this.store;
	if (isNil(client)) {
		throw new Error("no client in 'getMilestones'");
	} else {
		return get(
			"data",
			await client.get(`/projects/${projectId}/milestones`)
		);
	}
}

export { getMilestones2 };
