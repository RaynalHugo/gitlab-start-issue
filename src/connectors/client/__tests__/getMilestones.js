import { getMilestones2 } from "../getMilestones";
import { client } from "../createClient";

test("getMilestones2", async () => {
	expect(
		await getMilestones2(client, { projectId: 9112622 })
	).toMatchSnapshot();
});
