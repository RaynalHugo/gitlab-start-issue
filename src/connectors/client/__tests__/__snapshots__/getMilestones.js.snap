// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMilestones2 1`] = `
Array [
  Object {
    "created_at": "2018-11-20T13:24:50.424Z",
    "description": "",
    "due_date": "2019-11-29",
    "id": 716183,
    "iid": 14,
    "project_id": 9112622,
    "start_date": "2019-10-15",
    "state": "active",
    "title": "December 2019 Release",
    "updated_at": "2018-11-20T13:26:38.312Z",
    "web_url": "https://gitlab.com/sterblue/sterblue/milestones/14",
  },
  Object {
    "created_at": "2018-11-20T13:24:45.934Z",
    "description": "",
    "due_date": "2019-10-31",
    "id": 716182,
    "iid": 13,
    "project_id": 9112622,
    "start_date": "2019-09-16",
    "state": "active",
    "title": "November 2019 Release",
    "updated_at": "2018-11-20T13:27:18.998Z",
    "web_url": "https://gitlab.com/sterblue/sterblue/milestones/13",
  },
  Object {
    "created_at": "2018-11-20T13:24:40.978Z",
    "description": "",
    "due_date": "2019-09-30",
    "id": 716181,
    "iid": 12,
    "project_id": 9112622,
    "start_date": "2019-08-18",
    "state": "active",
    "title": "October 2019 Release",
    "updated_at": "2018-11-20T13:28:16.248Z",
    "web_url": "https://gitlab.com/sterblue/sterblue/milestones/12",
  },
  Object {
    "created_at": "2018-11-20T13:24:35.123Z",
    "description": "",
    "due_date": "2019-08-30",
    "id": 716180,
    "iid": 11,
    "project_id": 9112622,
    "start_date": "2019-07-15",
    "state": "active",
    "title": "September 2019 Release",
    "updated_at": "2018-11-20T13:28:38.082Z",
    "web_url": "https://gitlab.com/sterblue/sterblue/milestones/11",
  },
  Object {
    "created_at": "2018-11-20T13:24:28.843Z",
    "description": "",
    "due_date": "2019-07-30",
    "id": 716179,
    "iid": 10,
    "project_id": 9112622,
    "start_date": "2019-06-17",
    "state": "active",
    "title": "August 2019 Release",
    "updated_at": "2018-11-20T13:29:08.087Z",
    "web_url": "https://gitlab.com/sterblue/sterblue/milestones/10",
  },
  Object {
    "created_at": "2018-11-20T13:22:49.887Z",
    "description": "",
    "due_date": "2019-06-28",
    "id": 716178,
    "iid": 9,
    "project_id": 9112622,
    "start_date": "2019-05-15",
    "state": "active",
    "title": "July 2019 Release",
    "updated_at": "2018-11-20T13:29:35.408Z",
    "web_url": "https://gitlab.com/sterblue/sterblue/milestones/9",
  },
  Object {
    "created_at": "2018-11-19T18:55:33.954Z",
    "description": "",
    "due_date": "2019-05-30",
    "id": 715299,
    "iid": 8,
    "project_id": 9112622,
    "start_date": "2019-04-15",
    "state": "active",
    "title": "June 2019 Release",
    "updated_at": "2018-11-20T13:29:54.468Z",
    "web_url": "https://gitlab.com/sterblue/sterblue/milestones/8",
  },
  Object {
    "created_at": "2018-11-19T18:55:20.489Z",
    "description": "",
    "due_date": "2019-04-30",
    "id": 715298,
    "iid": 7,
    "project_id": 9112622,
    "start_date": "2019-03-15",
    "state": "active",
    "title": "May 2019 Release",
    "updated_at": "2018-11-20T13:30:19.261Z",
    "web_url": "https://gitlab.com/sterblue/sterblue/milestones/7",
  },
  Object {
    "created_at": "2018-11-19T18:55:02.412Z",
    "description": "",
    "due_date": "2019-03-29",
    "id": 715297,
    "iid": 6,
    "project_id": 9112622,
    "start_date": "2019-02-15",
    "state": "active",
    "title": "April 2019 Release",
    "updated_at": "2018-11-20T13:30:39.901Z",
    "web_url": "https://gitlab.com/sterblue/sterblue/milestones/6",
  },
  Object {
    "created_at": "2018-10-31T14:24:29.772Z",
    "description": "",
    "due_date": "2019-03-01",
    "id": 694535,
    "iid": 5,
    "project_id": 9112622,
    "start_date": "2019-01-15",
    "state": "active",
    "title": "March 2019 Release",
    "updated_at": "2018-10-31T14:24:29.772Z",
    "web_url": "https://gitlab.com/sterblue/sterblue/milestones/5",
  },
  Object {
    "created_at": "2018-10-31T14:22:14.617Z",
    "description": "",
    "due_date": "2019-01-29",
    "id": 694530,
    "iid": 4,
    "project_id": 9112622,
    "start_date": "2018-12-17",
    "state": "active",
    "title": "February 2019 Release",
    "updated_at": "2018-10-31T14:23:21.875Z",
    "web_url": "https://gitlab.com/sterblue/sterblue/milestones/4",
  },
  Object {
    "created_at": "2018-10-31T14:19:07.193Z",
    "description": "",
    "due_date": "2019-01-02",
    "id": 694528,
    "iid": 3,
    "project_id": 9112622,
    "start_date": "2018-11-15",
    "state": "active",
    "title": "January 2019 Release",
    "updated_at": "2018-11-22T09:57:16.595Z",
    "web_url": "https://gitlab.com/sterblue/sterblue/milestones/3",
  },
  Object {
    "created_at": "2018-10-31T14:17:59.700Z",
    "description": "",
    "due_date": "2018-11-29",
    "id": 694525,
    "iid": 2,
    "project_id": 9112622,
    "start_date": "2018-10-15",
    "state": "active",
    "title": "December 2018 Release",
    "updated_at": "2018-10-31T14:17:59.700Z",
    "web_url": "https://gitlab.com/sterblue/sterblue/milestones/2",
  },
  Object {
    "created_at": "2018-10-31T14:12:06.241Z",
    "description": "",
    "due_date": "2018-10-29",
    "id": 694515,
    "iid": 1,
    "project_id": 9112622,
    "start_date": "2018-09-16",
    "state": "closed",
    "title": "November 2018 Release",
    "updated_at": "2018-11-12T09:20:29.732Z",
    "web_url": "https://gitlab.com/sterblue/sterblue/milestones/1",
  },
]
`;
